engine/src/main/java/com/odysseusinc/arachne/executionengine/execution/r/RService.java

```diff
@@ -90,31 +90,38 @@ public Overseer analyze(AnalysisSyncRequestDTO analysis, File analysisDir, BiCon
                 return new FailedOverseer(Instant.now(), "Unable to resolve to [" + HOST_DOCKER_INTERNAL + "]", AnalysisRequestTypeDTO.R, e);
             }
         }
+
+        Path path = analysisDir.toPath();
+        AuthEffects auth = credentialProviders.stream().map(provider -> provider.apply(dataSource, path, path.toAbsolutePath().toString())).findFirst().orElse(null);
+
+        if (auth instanceof AuthEffects.ModifyUrl) {
+            String newUrl = ((AuthEffects.ModifyUrl) auth).getNewUrl();
+            dataSource.setConnectionString(newUrl);
+        }
+
```

Comment:
Is this still Java 8? If not, then it can be rewritten this way:
if (auth instanceof AuthEffects.ModifyUrl modifyUrl) {
    String newUrl = modifyUrl.getNewUrl();
    dataSource.setConnectionString(newUrl);
}

Reply:
Still on 8